test/unit/lexer/location_offset.d

```diff
@@ -0,0 +1,482 @@
+module lexer.location_offset;
+
+import dmd.lexer : Lexer;
+import dmd.tokens : TOK;
+
+import support : afterEach;
+import lexer.diagnostic_reporter : deinitializeFrontend;
+
+@("first token in the source code")
+unittest
+{
+    enum code = "token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 0, code);
+}
+
+@("first token when begoffset is not 0")
+unittest
+{
+    enum code = "ignored_token token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 13, code.length - 14, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 14, code);
+}
+
+@("last token in the source code")
+unittest
+{
+    enum code = "token1 token2 token3";
```

Comment:
When I mentioned "last token in the source code", I was more thinking of the offset of the last token would be the same as last byte. Something like `token1 token2 3` and check the offset of `3`.